src/hooks/auditoriumSections.ts

```diff
@@ -0,0 +1,134 @@
+import { User } from "types/User";
+
+import { setGridData } from "api/profile";
+
+import {
+  currentAuditoriumSectionsSelector,
+  currentAuditoriumSectionsByIdSelector,
+} from "utils/selectors";
+import { WithId } from "utils/id";
+import { getPositionHash } from "utils/auditorium";
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { useCallback, useMemo } from "react";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  sectionId: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const sections = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sections?.[sectionId];
+
+  const seatedUsers = useSectionSeatedUsers(venueId, sectionId);
+
+  const seatedUsersByHash = useMemo(
+    () =>
+      seatedUsers.reduce<Record<string, WithId<User> | undefined>>(
+        (acc, user) => {
+          if (!venueId) return acc;
+
+          const takenRow = user.data?.[venueId].row;
+          const takenColumn = user.data?.[venueId].column;
+
+          if (takenRow === undefined || takenColumn === undefined) return acc;
```

Comment:
Since I think I saw that we are using `null` for this data somewhere for some reason (which I suggested removing if possible), will this logic actually work as expected?

Reply:
It will work as expected since the whole gridOptions object is set to null.

Reply:
Would it be better to extract the `gridOptions` and check that then? Would make the code more obvious/aligned to how it's apparently implemented on the other side.

Something like:

```
const gridData = user.data?.[venueId];

if (!gridData) {
  return acc
}

const { row: takenRow, column: takenColumn } = gridData;
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1251/commits/adf6f1b18663d0e43209d78822eb78a1f85af80d